src/components/GoalActivityFeed.tsx

```diff
@@ -0,0 +1,191 @@
+                                        criteriaList={goalAchiveCriteria}
+                                        onAddCriteria={onGoalCriteriaAdd}
+                                        onToggleCriteria={onGoalCriteriaToggle}
+                                        onRemoveCriteria={onGoalCriteriaRemove}
+                                        onConvertToGoal={onGoalCriteriaConvert}
+                                        onUpdateCriteria={onGoalCriteriaUpdate}
+                                        onClick={onGoalCriteriaClick}
+                                        canEdit={_isEditable}
+                                        renderTrigger={(props) =>
+                                            nullable(_isEditable, () => (
+                                                <AddCriteriaForm
+                                                    goalId={props.goalId}
+                                                    onSubmit={props.onSubmit}
+                                                    validityData={props.validityData}
+                                                />
+                                            ))
+                                        }
+                                    />
+                                ))}
+
+                                {_relations.map((deps) =>
+                                    nullable(deps.goals.length || _isEditable, () => (
+                                        <GoalDependencyListByKind
+                                            goalId={id}
+                                            key={deps.kind}
+                                            kind={deps.kind}
+                                            items={deps.goals}
+                                            canEdit={_isEditable}
+                                            onRemove={onGoalDependencyRemove}
+                                            onClick={onGoalDependencyClick}
+                                        >
+                                            {nullable(_isEditable, () => (
+                                                <GoalDependencyAddForm
+                                                    onSubmit={onGoalDependencyAdd}
+                                                    kind={deps.kind}
+                                                    goalId={id}
+                                                    isEmpty={deps.goals.length === 0}
+                                                />
+                                            ))}
+                                        </GoalDependencyListByKind>
+                                    )),
+                                )}
+
+                                {nullable(lastStateComment, (value) => (
+                                    <CommentView
+                                        pin
+                                        id={value.id}
+                                        author={value.activity?.user}
```

Comment:
getUser? cc @LamaEats

Reply:
author is the User model, getUserName using by internal component

in other cases its the same